lambdas/src/apis/profiles/ensFiltering.ts

```diff
@@ -1,32 +1,32 @@
 import fetch from "node-fetch"
 
 const query = `
-  query GetNameByBeneficiary($beneficiary: String) {
-    nfts(where: { owner: $beneficiary, category: ens }) {
-      ens {
-        labelHash
-        beneficiary
-        caller
-        subdomain
-        createdAt
-      }
-    }
-  }`
+    query FilterNamesByOwner($owner: String, $names: [String]) {
+        nfts(
+            where: {
+                owner: $owner,
+                name_in: $names,
+                category: ens }) {
+            name
+        }
+    }`
 
-const opts = (ethAddress: string) => ({
+const opts = (ethAddress: string, names: string[]) => ({
     method: "POST",
     headers: { "Content-Type": "application/json" },
-    body: JSON.stringify({ query, variables: { beneficiary: ethAddress.toLowerCase() } })
+    body: JSON.stringify({ query, variables: {
+        owner: ethAddress.toLowerCase(),
+        names: names } })
 })
 
-export async function getOwnedENS(theGraphBaseUrl: string, ethAddress: string): Promise<string[]> {
+export async function filterENS(theGraphBaseUrl: string, ethAddress: string, namesToFilter: string[]): Promise<string[]> {
     const totalAttempts = 5
     for(let attempt=0; attempt<totalAttempts; attempt++) {
         try {
-            const response = await fetch(theGraphBaseUrl, opts(ethAddress))
+            const response = await fetch(theGraphBaseUrl, opts(ethAddress, namesToFilter))
```

Comment:
Maybe it would be a good time to replace this with our fetcher? Our fetcher supports both retries & timeouts

You can take a look at https://github.com/decentraland/catalyst/blob/master/content/src/service/access/AccessCheckerImpl.ts#L317